modin/data_management/data_manager.py

```diff
@@ -0,0 +1,1982 @@
+        
+        Args:
+            axis: The axis to extract the indices from.
+
+        Returns:
+            List of index names.
+        """
+        columns = list()
+        for col, dtype in zip(self.columns, self.dtypes):
+            if is_numeric_dtype(dtype):
+                columns.append(col)
+        return columns
+
+    def numeric_function_clean_dataframe(self, axis):
+        """Preprocesses numeric functions to clean dataframe and pick numeric indices.
+        
+        Args:
+            axis: '0' if columns and '1' if rows.
+
+        Returns:
+            Tuple with return value(if any), indices to apply func to & cleaned Manager.
+        """
+        result = None
+        index = self.numeric_columns()
+        data_manager = self
+
+        # If no numeric columns and over columns, then return empty Series
+        if not axis and len(index) == 0:
+            result = pandas.Series(dtype=np.float64)
+
+        if axis:
+            nonnumeric = [col for col, dtype in zip(self.columns, self.dtypes) if not is_numeric_dtype(dtype)]
+            if len(nonnumeric) == len(self.columns):
+                # If over rows and no numeric columns, return this
+                result = pandas.Series([np.NaN for _ in self.index])
+            else:
+                data_manager = self.drop(columns=nonnumeric)
+                index = data_manager.index
+
+        return (result, index, data_manager)
+    # END Internal methods
+
+    # Metadata modification methods
+    def add_prefix(self, prefix):
+        new_column_names = self.columns.map(lambda x: str(prefix) + str(x))
+        return self.__constructor__(self.data, self.index, new_column_names, self._dtype_cache)
+
+    def add_suffix(self, suffix):
+        new_column_names = self.columns.map(lambda x: str(x) + str(suffix))
+        return self.__constructor__(self.data, self.index, new_column_names, self._dtype_cache)
+    # END Metadata modification methods
+
+    # Copy
+    # For copy, we don't want a situation where we modify the metadata of the
+    # copies if we end up modifying something here. We copy all of the metadata
+    # to prevent that.
+    def copy(self):
+        return self.__constructor__(self.data.copy(), self.index.copy(), self.columns.copy(), self._dtype_cache)
+
+    # Append/Concat/Join (Not Merge)
+    # The append/concat/join operations should ideally never trigger remote
+    # compute. These operations should only ever be manipulations of the
+    # metadata of the resulting object. It should just be a simple matter of
+    # appending the other object's blocks and adding np.nan columns for the new
+    # columns, if needed. If new columns are added, some compute may be
+    # required, though it can be delayed.
+    #
+    # Currently this computation is not delayed, and it may make a copy of the
+    # DataFrame in memory. This can be problematic and should be fixed in the
+    # future. TODO: Delay reindexing
```

Comment:
Can you sign it? Like TODO(devin)

Reply:
sure i can do that

Reply:
also, all of these TODOs should be issues, i'll create the ones that I wrote